### PR TITLE
a quickstart guide on bot development

### PR DESCRIPTION
**Pull request by missytake**
related: #44

### CODE REVIEW

File: quickstart.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+The core engine can be used natively from rust, and through a [C-API](https://c.delta.chat) from C and C++. (Please note that the rust API is still subject to change, while the C-API is considered stable.)
+
+For the following languages there are bindings to the C-API:
+* Python: [https://py.delta.chat](https://py.delta.chat)
+* NodeJS: [https://github.com/deltachat/deltachat-node](https://github.com/deltachat/deltachat-node)
+* Golang: [https://github.com/hugot/go-deltachat](https://github.com/hugot/go-deltachat)
+
+Additionally there are two small libraries for NodeJS that provide a higher level API to write bots:
+* [deltachat-node-bot-base](https://github.com/deltachat-bot/deltachat-node-bot-base) — for writing bots (duh!),
+* [deltachat-node-webbot-base](https://github.com/deltachat-bot/deltachat-node-webbot-base) — for web-interacting bots (bases on the former).
```
Comment (Simon-Laux) [inline quickstart.md]: don't mention them or update them, they are waay outdated

File: quickstart.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+In Python this might look like this (with a configured account and running threads):
+```python
+while 1:
+    event = account._evlogger.get_matching("DC_EVENT_INCOMING_MSG")
+    if event[2] != 0:
+        msg = account.get_message_by_id(event[2])
+        maybe_do_something(msg)
+```
+
+Messages that don't belong to an existing chat must be filtered from a different event: [`DC_EVENT_MSGS_CHANGED`](https://c.delta.chat/group__DC__EVENT.html#ga0f52cdaad70dd24f7540abda6193cc2d). The code must check that the message is put into the chat called "deaddrop", which is the bucket for all content messages that don't belong to an existing chat yet (e.g. if someone sends you an initial message).
```
Comment (Simon-Laux) [inline quickstart.md]: deaddrops is not the right term anymore it's contact requests, and bots auto accept them now when they set the config flag `bot` to `true`/`1` https://github.com/deltachat-bot/echo/issues/33

File: quickstart.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+[`account.run_account()`](https://py.delta.chat/api.html#deltachat.account.Account.run_account)
+will run the account, and configure it if necessary (e.g. look up the mail
+servers of your provider).
+
+[`account.wait_shutdown()`](https://py.delta.chat/api.html#deltachat.account.Account.wait_shutdown)
+waits until the account quits running for some reason. This is necessary here
+so the script doesn't quit directly, but stays open as long as the bot is
+running. We want the bot to reply to our messages forever, after all.
```
Comment (Simon-Laux) [inline quickstart.md]: maybe move the text to comments and just have a list of links after it? If we even need the links to the code reference?

File: quickstart.md (hunk 1 above)
Comment (missytake) [reply quickstart.md]: ah yes, everything below `## API` will be removed later, that's just the rest of "development.html" what I copied before committing^^

File: quickstart.md (hunk 2 above)
Comment (missytake) [reply quickstart.md]: ah yes, everything below `## API` will be removed later, that's just the rest of "development.html" what I copied before committing^^

File: quickstart.md (hunk 3 above)
Comment (missytake) [reply quickstart.md]: They are now not directly after each other, but are explaining a code snippet directly - I think it makes sense now as it is.

**Comment by missytake** [comment]
The page can be previewed here: https://staging.bots.delta.chat/45/quickstart.html

File: quickstart.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+**Tip:** You can look at the [python
+docs](https://py.delta.chat/api.html#deltachat.message.Message) to see what
+else you can do with a deltachat.Message object.
+{: .notification }
+
+If we use `ac.get_fresh_messages()` to fetch new messages, we need to mark the
+messages as seen afterwards, so we don't re-fetch it every time.
+
+This generally works, but it is a bit primitive. You can also use event hooks:
```
Comment (link2xt) [inline quickstart.md]: This does not explain why this approach is bad. Primitive does not mean it's bad. I'd rather explain that this eats 100% constantly checking for fresh messages, that is why we want an event-driven approach instead.

**Comment by link2xt** [comment]
@missytake  This does not have to wait for https://github.com/deltachat/deltachat-core-rust/pull/3734. Except for the need to know about async, I like the approach in https://github.com/deltachat/deltachat-core-rust/pull/3734 better because there is a single loop that processes all the events and messages. But it is not applicable to CFFI python bindings because they already implement their own event loop (`_inner_run`), and the only thing you can do is hooking into it.

**Review by link2xt — APPROVED** [review]

**Pull request merged**